modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -185,10 +189,23 @@ inline std::vector<int> toCV(const std::vector<int64_t> &vsz) {
 }
 
 inline cv::Mat toCV(Ort::Value &v) {
-    auto info = v.GetTensorTypeAndShapeInfo();
-    return cv::Mat(toCV(info.GetShape()),
-                   toCV(info.GetElementType()),
-                   reinterpret_cast<void*>(v.GetTensorMutableData<uint8_t*>()));
+    const auto info = v.GetTensorTypeAndShapeInfo();
+    int type = -1;
+    bool is_converted = false;
+    std::tie(type, is_converted) = toCV(info.GetElementType());
+    const auto shape = toCV(info.GetShape());
+    if (!is_converted) {
+        return cv::Mat(shape,
+                       type,
+                       reinterpret_cast<void*>(v.GetTensorMutableData<uint8_t*>()));
+    }
+    const int total = std::accumulate(shape.begin(), shape.end(), 1, std::multiplies<int>());
+    cv::Mat mt(shape, CV_32S);
+    int64_t* ptr = v.GetTensorMutableData<int64_t>();
+    int* mt_ptr = mt.ptr<int>();
+    std::transform(ptr, ptr + total, mt_ptr,
+                   [](int64_t el) { return static_cast<int>(el); });
```

Comment:
hmm what about strides? I'd put an assert here expecting a Mat is continuous.

Reply:
Added.

Reply:
It makes sense to place checks before related statements.

Reply:
Changed. But i don't see where gap can appear in Mat for current implementation.